examples/expressions/main.rs

```diff
@@ -0,0 +1,84 @@
+mod metadata;
+mod note;
+
+use gtk::gio;
+use gtk::glib;
+use gtk::prelude::*;
+
+use self::metadata::Metadata;
+use self::note::Note;
+
+fn main() {
+    let application = gtk::Application::new(
+        Some("com.github.gtk-rs.examples.expressions"),
+        Default::default(),
+    );
+
+    application.connect_activate(build_ui);
+    application.run();
+}
+
+fn build_ui(app: &gtk::Application) {
+    let window = gtk::ApplicationWindow::new(app);
+    window.set_title(Some("Expressions"));
+    window.set_default_size(300, 500);
+
+    let factory = gtk::SignalListItemFactory::new();
+    factory.connect_setup(|_, list_item| {
+        let hbox = gtk::Box::new(gtk::Orientation::Horizontal, 24);
+
+        let title_label = gtk::Label::new(None);
+        hbox.append(&title_label);
+
+        let last_modified_label = gtk::Label::new(None);
+        hbox.append(&last_modified_label);
+
+        // Instead of binding properties and manually unbinding them, you can create expressions.
+        // The value can be obtained even if it is several steps away.
+        let metadata_expression = list_item
+            .property_expression("item")
+            .chain_property(Note::static_type(), "metadata");
+
+        metadata_expression
+            .chain_property(Metadata::static_type(), "title")
+            .chain_closure(|args| {
+                let title: String = args[1].get().unwrap();
+                format!("Title: {}", title)
+            })
+            .bind(&title_label, "label", None::<&gtk::Widget>);
+
+        metadata_expression
+            .chain_property(Metadata::static_type(), "last-modified")
+            .chain_closure(|args| {
+                let last_modified: glib::DateTime = args[1].get().unwrap();
+                format!("Last Modified: {}", last_modified.format_iso8601().unwrap())
+            })
+            .bind(&last_modified_label, "label", None::<&gtk::Widget>);
+
+        list_item.set_child(Some(&hbox));
+    });
+
+    let model = gtk::NoSelection::new(Some(&data()));
+    let list_view = gtk::ListView::new(Some(&model), Some(&factory));
+
+    let scrolled_window = gtk::ScrolledWindow::new();
+    scrolled_window.set_child(Some(&list_view));
+    scrolled_window.set_policy(gtk::PolicyType::Never, gtk::PolicyType::Automatic);
+
+    window.set_child(Some(&scrolled_window));
+    window.show();
+}
+
+fn data() -> gio::ListStore {
+    let list_store = gio::ListStore::new(Note::static_type());
+
+    for i in 0..100 {
+        let metadata = Metadata::new();
+        metadata.set_property("title", format!("Note ({})", i));
+
+        let note = Note::new(&metadata);
+        list_store.append(&note);
```

Comment:
You will want to fill a vector and then add it to the list store in one go in order to prevent 100 signal emissions.
See e.g. https://github.com/gtk-rs/gtk4-rs/blob/f14d7eb1edbbd33b9de2820f5f88a9a52688be96/book/listings/list_widgets/5/main.rs#L34